.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,157 @@
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Don't save Bash session history
+        run: unset HISTFILE
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          check-latest: true
+
+      - name: Configure kernel (increase watchers)
+        run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+
+      # this step might be replaced with a reusable workflow get-variables.yml (see above)
+      - name: Get variables (branch name, commit hash, yarn cache directory)
+        id: get-variables
+        run: |
+          echo "::set-output name=shortref::${GITHUB_REF#refs/*/}";
+          echo "::set-output name=commitsha::$(echo ${GITHUB_SHA})";
+          echo "::set-output name=yarncachedir::$(yarn cache dir)";
+
+      - name: Workspace cache
+        uses: actions/cache@v3
+        id: workspace-cache
+        env:
+          cache-name: workspace-cache
+        with:
+          path: |
+            ~/.cache
+            ./node_modules
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ steps.get-variables.outputs.shortref }}-workspace-${{ steps.get-variables.outputs.commitsha }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ steps.get-variables.outputs.shortref }}-workspace-
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-
+            ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+            ${{ runner.os }}-
+          ## the above might be replaced with the commented code beneath if a reusable workflow get-variables.yml is used (see above)
+          # key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ needs.get-variables.outputs.shortref }}-workspace-${{ needs.get-variables.outputs.commitsha }}
+          # restore-keys: |
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ needs.get-variables.outputs.shortref }}-workspace-
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+          #   ${{ runner.os }}-
+
+      - name: Install project dependencies
+        run: yarn install --frozen-lockfile --non-interactive
+
+      # Is 'yarn pack --filename ./dist/ngxs-core.tgz' needed during premerge checks?
+      - name: Compile NGXS and create a pack
+        run: |
+          yarn build
+          yarn pack --filename ./dist/ngxs-core.tgz
+
```

Comment:
I see that jobs run in parallel by default and you can control sequencing through the [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) property.
I think steps can only be sequential.

It would be ideal if all the linting and testing steps after the build were other jobs (declaring a `needs: build` ) so that they can run in parallel, but wait for the build job first

I think that all of these testing/checking steps may be a good candidate for the use of the matrix job, where the script run is the varied part of the job.